src/Akeneo/Crowdin/TranslatedProgressSelector.php

```diff
@@ -43,36 +60,102 @@ public function __construct(Client $client, EventDispatcherInterface $eventDispa
+                true
+            );
+            $table = new Table($output);
+            $table->setHeaders(['locale', 'percentage']);
+            foreach ($this->packages(true, $branch) as $code => $progress) {
+                $table->addRow([$code, sprintf('%d%%', $progress)]);
+            }
+            $table->render();
         }
     }
 
     /**
      * Return the list of packages to import
      *
+     * @param bool        $exclude
+     * @param string|null $branch
+     *
      * @return array
      */
-    public function packages()
+    public function packages($exclude = true, $branch = null)
     {
         $this->eventDispatcher->dispatch(Events::PRE_CROWDIN_PACKAGES);
 
-        $response = $this->client->api('status')->execute();
-        $xml = simplexml_load_string($response);
-        $codesToImport = [];
+        $maxApproved = -1;
+        $approvedCounts = [];
+        foreach ($this->getAllCrowdinCodes() as $code) {
+            $approved = $this->getApprovedCount($code, $branch);
+            $approvedCounts[$code] = $approved;
+            $maxApproved = max($maxApproved, $approved);
+        }
 
-        foreach ($xml as $xmlElement) {
-            $translated_progress = (int) $xmlElement->translated_progress;
-            if ($translated_progress >= $this->minTranslatedProgress) {
-                $code = (string) $xmlElement->code;
-                $codesToImport[$code] = $translated_progress;
+        $result = [];
+        $minApprovedCount = $maxApproved * $this->minTranslatedProgress / 100;
+        foreach ($approvedCounts as $code => $approvedCount) {
+            if (!$exclude || $approvedCount > $minApprovedCount) {
+                $result[$code] = $approvedCount / $maxApproved * 100;
             }
         }
 
+        asort($result);
+        $result = array_reverse($result);
+
         $this->eventDispatcher->dispatch(Events::POST_CROWDIN_PACKAGES, new GenericEvent($this, [
-            'count' => count($codesToImport),
+            'count' => count($result)
         ]));
 
-        return $codesToImport;
+        return $result;
+    }
+
+    /**
+     * Returns the count of approved strings for a specific language code.
+     *
+     * @param string      $crowdinCode
+     * @param null|string $branch
+     *
+     * @return int
+     */
+    protected function getApprovedCount($crowdinCode, $branch = null)
+    {
+        $query = $this->client->api('language-status');
+        $query->setLanguage($crowdinCode);
+        $response = $query->execute();
+
+        $xml = simplexml_load_string($response);
+
+        $approved = 0;
+        foreach ($xml->files->item as $mainNode) {
+            if ((null !== $branch) && ('branch' === (string) $mainNode->node_type) && ($branch === (string) $mainNode->name)) {
+                foreach ($mainNode->files->item as $mainDir) {
+                    if (null === $this->folders || in_array((string) $mainDir->name, $this->folders)) {
```

Comment:
is it normal we use $this->folders just to count the number of approved messages?

if we don't use it to find the total number of messages, I guess the result is different no?
maybe I miss something